utils/slice.go

```diff
@@ -10,3 +17,11 @@ func IndexOf(strs []string, str string) int {
 
 	return -1
 }
+
+// Nonce calculates a byte slice using the context's tx bytes and gas meter
+func Nonce(ctx sdk.Context) [sha256.Size]byte {
```

Comment:
I actually think this is the most adequate place for this function, because the bitcoin module will not be the only one using it (the other ones will be EVM and axelarnet). I also don't see the bit issue with passing sdk.Context.

Reply:
also, keep in mind that the idea is to generate the nonce deterministically, and sdk.Context is a pretty good source for deterministic data we can use as seed. Furthermore, by providing the sdk.Context instead of more specific parameters, it will also be easier to change the implementation if we ever need to do that in the future.

Reply:
@sammy1991106 do you have an alternative suggestion for location. I agree on the context, pass in the tx bytes and consumed gas directly. That would also make it SOOO much easier to test the function

Reply:
It might make it easier to test the function, but not other functions that use it internally. You would still need to build the context appropriately. So I don't see many benefits in changing the parameters for this function.

Reply:
why is it hard for internal functions to use otherwise? It makes it explicit what this function exactly depends on

Reply:
You said that giving the exact arguments to `Nonce` instead of the whole `sdk.Context` object makes it easier to test the function, which is true. But for example `GetBurnerAddressAndSalt` will still invoke the function from within.  So when testing `GetBurnerAddressAndSalt` (and other functions that also invoke `Nonce` internally and receive a context) you still need to assemble an appropriate `sdk.Context` object when creating unit tests. So in terms of unit testing, I don't see a lot of benefit.

Reply:
I still think there are more advantages with providing the whole context object, but I changed the function signature to receive an hash and the gas meter